File: packages/ios/Avocado/Avocado/Plugins/Console.swift
import Foundation

public class Console : Plugin {

  @objc public func log(_ call: PluginCall) {
    let data = call.options
    let message = data["message"] ?? ""
    let level = data["level"] ?? "LOG"
    print("[\(level)] \(self.pluginId) - \(message)")
  }
}

File: android/avocado/src/main/java/com/avocadojs/plugin/Console.java
package com.avocadojs.plugin;

import android.util.Log;

import com.avocadojs.NativePlugin;
import com.avocadojs.Plugin;
import com.avocadojs.PluginCall;
import com.avocadojs.PluginMethod;


@NativePlugin(id="com.avocadojs.plugin.console")
public class Console extends Plugin {

  @PluginMethod()
  public void log(PluginCall call) {
    String level = call.getString("level", "log").toLowerCase();
    String message = call.getString("message", "");

    if (level == "error'") {
      Log.e("console", message);

    } else if (level == "warn'") {
      Log.w("console", message);

    } else if (level == "debug'") {
      Log.d("console", message);

    } else {
      Log.i("console", message);
    }
  }

}
